language/move-compiler/src/parser/lexer.rs

```diff
@@ -224,8 +224,7 @@ impl<'input> Lexer<'input> {
         // a multi-line or single-line comment.
         loop {
             // Trim only the whitespace characters we recognize: newline(\n|\r\n), tab, and space.
-            text = text.trim_start_matches("\r\n");
-            text = text.trim_start_matches(|c: char| matches!(c, '\n' | '\t' | ' '));
+            text = text.trim_start_matches(|c: char| matches!(c, '\r' | '\n' | '\t' | ' '));
```

Comment:
As far as I see this would again allow us to accept arbitrary CR in the source, which we do not want, Your trimming algorithm need to be smarter and handwritten. Something like:

```rust
fn move_trim(s: &str) -> String {
  let mut i = 0;
  while i < s.len() {
    match s[i] {
      `\t' | '\r' |  '\n` => i += 1,
      `\r' if i > 0 && s[i-1] == `\n` => i += 1,
      _ => break
    }
  }
  s[i..].to_string()
} 
```

Reply:
@wrwg If verify_string has ensured that CRLF is present at the same time, then trim_whitespace_and_comments will not have a separate CR here.

Reply:
Hey Owen, can you use a link to the source in your github clone instead of a screenshot? This would make it easier to verify this.

You may not be aware of this, but a few years ago, Diem had a significant problem with external auditors pointing out security issues with the lexer, the code you are modifying. This was back then a public relations disaster. That is why we are today very, very careful about which characters we allow. An auditor may come along trim your code and don't see your verify_string. So for double extra safety, make the trim function secure to only trim CRLF and not standalone CR. Thank you.

Reply:
Ok, thank you very much for the tip. I will pay particular attention to the security of the submitted PR later.

Reply:
I'll replace the screenshots later